CRM/Xdedupe/Resolver/CustomGroupIntegral.php

```diff
@@ -0,0 +1,185 @@
+<?php
+/*-------------------------------------------------------+
+| SYSTOPIA's Extended Deduper                            |
+| Copyright (C) 2025 SYSTOPIA                            |
+| Author: B. Endres (endres@systopia.de)                 |
+|         S. Frank (frank@systopia.de)                   |
+| http://www.systopia.de/                                |
++--------------------------------------------------------+
+| This program is released as free software under the    |
+| Affero GPL license. You can redistribute it and/or     |
+| modify it under the terms of this license which you    |
+| can read by viewing the included agpl.txt or online    |
+| at www.gnu.org/licenses/agpl.html. Removal of this     |
+| copyright header is strictly prohibited without        |
+| written permission from the original author(s).        |
++--------------------------------------------------------*/
+
+declare(strict_types = 1);
+use CRM_Xdedupe_ExtensionUtil as E;
+
+/**
+ * Implements a resolver for contact custom group conflicts.
+ *
+ * This version will treat the custom group data
+ * in an integral way, i.e. the fields of the group should be treated
+ * as one record set
+ */
+class CRM_Xdedupe_Resolver_CustomGroupIntegral extends CRM_Xdedupe_Resolver {
+
+  /**
+   * @var integer ID of the custom group
+   */
+  protected int $custom_group_id;
+
+  protected ?array $custom_group_data = null;
+
+  /**
+   * @param CRM_Xdedupe_Merge $merge
+   * @param mixed $custom_group_id
+   */
+  public function __construct($merge, $custom_group_id) {
+    $this->custom_group_id = (int) $custom_group_id;
+    parent::__construct($merge);
+  }
+
+  /**
+   * Get the spec (i.e. class name) that refers to this resolver
+   * @return string spec
+   */
+  public function getSpec() {
+    return "CRM_Xdedupe_Resolver_CustomGroupIntegral:{$this->custom_group_id}";
+  }
+
+  /**
+   * Report the contact attributes that this resolver requires
+   *
+   * @return array<string> list of contact attributes
+   */
+  public function getContactAttributes() {
+    return ['id'];
+  }
+
+  /**
+   * get the name of the finder
+   * @return string name
+   */
+  public function getName() {
+    return E::ts("\"%1\" conflicts (integral)", [
+            1 => $this->getCustomGroupData()['title']]);
+  }
+
+    /**
+     * Get information on the custom group we're working on
+     *
+     * @return array custom group data
+     * @throws CRM_Core_Exception
+     */
+  public function getCustomGroupData() : array {
+      if ($this->custom_group_data === null) {
+          $this->custom_group_data = civicrm_api4(
+                  'CustomGroup', 'get',
+                  [
+                          'select' => ['title', 'table_name'],
+                          'where' => [['id', '=', $this->custom_group_id]],
+                  ]
+          )->first();
```

Comment:
Should be `single()` instead of `first()` because it is an error here if there's no result.